components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LoginResource.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.micro.integrator.management.apis;
+
+import com.nimbusds.jose.JOSEException;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseConstants;
+import org.apache.synapse.config.SynapseConfiguration;
+import org.json.JSONObject;
+import org.wso2.micro.integrator.management.apis.security.handler.*;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+public class LoginResource implements MiApiResource {
+
+    private static final Log LOG = LogFactory.getLog(LoginResource.class);
+
+    // HTTP method types supported by the resource
+    Set<String> methods;
+
+    public LoginResource() {
+        methods = new HashSet<>();
+        methods.add(Constants.HTTP_POST);
+        methods.add(Constants.HTTP_GET);
+    }
+
+    @Override
+    public Set<String> getMethods() {
+        return methods;
+    }
+
+    @Override
+    public boolean invoke(MessageContext messageContext, org.apache.axis2.context.MessageContext axis2MessageContext,
+                          SynapseConfiguration synapseConfiguration) {
+        //Init token store
+        JWTTokenStore tokenStore = JWTInMemoryTokenStore.
+                getInstance(JWTConfig.getInstance().getJwtConfigDto().getTokenStoreSize());
+
+        //UUID used as unique token
+        UUID uuid = UUID.randomUUID();
+        String randomUUIDString = uuid.toString();
+
+        JWTTokenInfoDTO newToken = new JWTTokenInfoDTO();
+        newToken.setToken(randomUUIDString);
+        newToken.setScope(AuthConstants.JWT_TOKEN_DEFAULT_SCOPE);
+        newToken.setIssuer((String) axis2MessageContext.getProperty(SynapseConstants.SERVER_IP));
+        Long time = new Date().getTime();
+        String expiryConfig = JWTConfig.getInstance().getJwtConfigDto().getExpiry();
+        Long expiryInMins = AuthConstants.DEFAULT_EXPIRY_DURATION;
+        if(expiryConfig != null && !expiryConfig.isEmpty()) {
```

Comment:
Can we use StringUtils.isEmpty(expiryConfig) for this?

Reply:
Thanks fixed with 19b1070